src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/CapiHelper.cs

```diff
@@ -69,24 +70,32 @@ public static string UpgradeRSA(int dwProvType, string wszProvider)
             bool requestedEnhanced = string.Equals(wszProvider, MS_ENHANCED_PROV, StringComparison.Ordinal);
             bool requestedBase = string.Equals(wszProvider, MS_DEF_PROV, StringComparison.Ordinal);
             string wszUpgrade = null;
-            SafeProvHandle safeProvHandle = SafeProvHandle.InvalidHandle;
             if (requestedBase || requestedEnhanced)
             {
+                SafeProvHandle safeProvHandle;
+
                 // attempt to use the AES provider
-                if (S_OK == AcquireCryptContext(ref safeProvHandle, null, MS_ENH_RSA_AES_PROV,
+                if (S_OK == AcquireCryptContext(out safeProvHandle, null, MS_ENH_RSA_AES_PROV,
                                                 dwProvType, (uint)CryptAcquireContextFlags.CRYPT_VERIFYCONTEXT))
                 {
                     wszUpgrade = MS_ENH_RSA_AES_PROV;
                 }
+
+                safeProvHandle.Dispose();
             }
-            else if (wszUpgrade == null && requestedBase)
+
+            if (wszUpgrade == null && requestedBase)
```

Comment:
Actually, not only is it likely not worthwhile on Core (since PROV_RSA_AES should always work); I just double-checked the legacy code... and this is also dead code on Desktop.  I'll just delete this clause, entirely.

Good catch.